README.md

```diff
@@ -1,36 +1,312 @@
+    cert: "path/to/apns/cert.pem", # mandatory, path to certificate file provided by APNS
+    key: "path/to/apns/key.pem", # mandatory, path to key file provided by APNS
+    # Connection config
+    endpoint: "api.push.apple.com", # optional
+    port: 443, # optional
+    tls_opts: [], # optional
+    ping_interval: 5000, # optional
+    reconnect_attempts: 3, # optional
+    # pool config
+    tags: [:another_batch_clients], # optional
+    worker_num: 3, # optional
+    raw_opts: [] # optional
+]
 ```
 
-Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_doc)
-and published on [HexDocs](https://hexdocs.pm). Once published, the docs can
-be found at [https://hexdocs.pm/sparrow](https://hexdocs.pm/sparrow).
+### APNS token example
+
+```elixir
+apns_token = [
+          token_id: :some_atom_id, # mandatory, the same as in APNSPOOL
+          key_id: "FAKE_KEY_ID", # mandatory, data obtained form APNS account
+          team_id: "FAKE_TEAM_ID", # mandatory, data obtained form APNS account
+          p8_file_path: "path/to/file/token.p8" # mandatory, path to file storing APNS token
+        ]
+```
+
+## Include sparrow in your project
+
+```elixir
+defp deps do
+    [
+      {:sparrow, github: "esl/sprrow", tag: "cc80bbc"},
+      ]
+  end
+```
+
+## Many pools
+
+Sparrow is supporting many pools in a single `:sparrow` instance. 
+[Tags](#Tags) are used to choose pool when sending a notification.
+[](https://placehold.it/15/ff0000/ff0000?text=+) *`This feature is not yet implemented for FCM.`*
+
+## Tags
+
+Tags is a mechanism allowing to choose a pool to send a notification.
+Each pool has a defined list of tags (`[]` as default).
+Algorithm is:
+1) Filter pooltype based on notification type (`:fcm` or `:{apns, :dev}` or `{:apns, :prod}`).
+2) Choose only pools that have all tags (from the function call) included in their tags (from pool configuration).
+3) Choose first of the filtered pools.
+
+Example:
+    Let's say you have following pools:
+
+- `{:apns, :dev}`:
+    - *pool1*: `[:test_pool, :dev_pool, :homer]`,
+    - *pool2*: `[:test_pool, :dev_pool, :bart]`,
+    - *pool3*: `[:test_pool, :dev_pool, :ned, :bart]`
+- `{:apns, :prod}`:
+    - *pool1*: `[:prod_pool]`
+
+
+Lets assume notification type is `{:apns, :dev}`.
+
+If you pass `[]`, *pool1* is chosen. 
+
+If you pass `[:homer]`, *pool1* is chosen. 
+
+If you pass `[:bart]`, *pool2* is chosen. 
+
+If you pass `[:ned]`, *pool3* is chosen. 
+
+If you pass `[:test_pool]`, *pool1* is chosen. 
+
+If you pass `[:test_pool, :homer]`, *pool1* is chosen. 
+
+If you pass `[:test_pool, :dev_pool, :homer]`, *pool1* is chosen. 
+
+If you pass `[:test_pool, :dev_pool, :ned]`, *pool3* is chosen. 
+
+If you pass `[:not_existing, :set_of_tags]`, `{:error, :configuration_error}` is returned. 
+
+[](https://placehold.it/15/ff0000/ff0000?text=+) *`It is not recommended to choose pool based on pools order!`*
```

Comment:
[](https://placehold.it/15/ff0000/ff0000?text=+) *`It is not recommended to choose a pool based on pools order!`*